pageserver/src/pgdatadir_mapping.rs

```diff
@@ -196,6 +198,45 @@ impl<R: Repository> DatadirTimeline<R> {
         Ok(exists)
     }
 
+    pub fn lookup_lsn(
+        &self,
+        timestamp_opt: Option<TimestampTz>,
+        xid_opt: Option<TransactionId>,
```

Comment:
I will add comment here. We have some discussion about it:
https://github.com/zenithdb/zenith/issues/1320#issuecomment-1054115618
So this is not a question of whether there will be one or two functions.
The question is how our users are going to specify PITR interval?
Should be allow to specify both LSN range and time duration? (OR/AND)
Or user need to choose one of them?
We should also somehow limit PITR for free-tier. Limiting it only by time (for example 7 days)
may require too larger storage overhead.
From the other side, saying that we will store 15Gb of WAL is not a user friendly information: he will have no idea how long in the past he can perform recovery.

The safest criteria for users is AND (current choice): we can drop delta layer only if it is out of PITR LSN range and PITER time interval.
The safest criteria for us is OR: we will keep data for 7 days but only if it doesn't exceed 15Gb.

So it seems to be more marketing than technical question.

Reply:
I agree that 15GB WAL limit is not an intuitive concept to expose to app developers, and it's almost unacceptable for a developer centered product. But I see it more as a technical preview limitation, or a placeholder, rather than a real long-term product decision. We should give it more thought. The topic of PITR from s3 is also relevant, since not only is the 15GB limit unintuitive, it's also pretty small.

I'm sure this PR will land before the correct product decision is made. So I suggest just documenting the current behavior at least, and possibly linking to any issue/discussion about improvements.

Reply:
Sorry, looks like I have answered to some other question:(
What I have wrote relates to definition of PITR interval at pageserver, but not to LSM mapping.
In any case I will add comment explaining `lookup_lsn` function behavior.